src/org/labkey/test/pages/assay/AssayRunsPage.java

```diff
@@ -55,6 +55,27 @@ public AssayDataPage clickAssayIdLink(String assayId)
         return new AssayDataPage(getDriver());
     }
 
+    public UpdateQCStatePage updateSelectedQcStatus()
+    {
+        getTable().clickHeaderMenu("QC State", "Update state of selected rows");
+        return new UpdateQCStatePage(getDriver());
+    }
+
+    public ManageAssayQCStatesPage manageQCStates()
+    {
+        getTable().clickHeaderMenu("QC State", "Manage states");
+        return new ManageAssayQCStatesPage(getDriver());
+    }
+
+    public AssayRunsPage setRowQcStatus(int rowIndex, String state, String comment)
+    {
+        getTable().checkCheckbox(rowIndex);
+        return updateSelectedQcStatus()
+                .selectState(state)
+                .setComment(comment)
+                .clickUpdate();
+    }
```

Comment:
Have this take multiple row indexes, to prevent against unnecessary page loads when setting QC state on multiple rows.
`setRowQcStatus(String state, String comment, Integer... rowIndices)`

Reply:
okay.